arex-schedule-web-api/src/main/java/com/arextest/schedule/planexecution/PlanStageMonitorImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.arextest.schedule.planexecution;
+
+import com.arextest.schedule.dao.mongodb.ReplayPlanRepository;
+import com.arextest.schedule.mdc.MDCTracer;
+import com.arextest.schedule.model.ReplayPlan;
+import com.arextest.schedule.model.plan.PlanStageEnum;
+import com.arextest.schedule.model.plan.ReplayPlanStageInfo;
+import com.arextest.schedule.model.plan.StageStatusEnum;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Resource;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * @author wildeslam.
+ * @create 2023/7/25 20:03
+ */
+@Slf4j
+@Component
+public class PlanStageMonitorImpl implements PlanExecutionMonitor {
```

Comment:
Can this be done in the existing execution monitor?
adding another internal class should be enough

Reply:
I thought so too at first. But the calling timing of register&deregister function for two monitors are different.